src/main/java/roomescape/service/ReservationService.java

```diff
@@ -1,36 +1,44 @@
 package roomescape.service;
 
-import org.springframework.transaction.annotation.Transactional;
-import roomescape.dao.ReservationDAO;
+import roomescape.exception.DuplicateReservationScheduleException;
+import roomescape.exception.ReservationNotFoundException;
+import roomescape.exception.ThemeNotFoundException;
+import roomescape.repository.ReservationRepository;
+import roomescape.repository.ThemeRepository;
 import roomescape.domain.Reservation;
 import org.springframework.stereotype.Service;
-import roomescape.dao.ReservationAppDAO;
-import roomescape.dto.ReservationRequest;
+import roomescape.domain.Theme;
+import roomescape.dto.ReservationCreateRequest;
+
+import java.util.NoSuchElementException;
 
 @Service
-@Transactional
 public class ReservationService {
 
-    private ReservationDAO reservationDAO;
+    private final ReservationRepository reservationRepository;
+    private final ThemeRepository themeRepository;
 
-    public ReservationService(ReservationAppDAO reservationAppDAO) {
-        this.reservationDAO = reservationAppDAO;
+    public ReservationService(ReservationRepository reservationRepository, ThemeRepository themeRepository) {
+        this.reservationRepository = reservationRepository;
+        this.themeRepository = themeRepository;
     }
 
-    public Reservation createReservation(ReservationRequest reservationRequest) {
-        if (reservationDAO.checkSchedule(reservationRequest) == 0) {
-            Long id = reservationDAO.addReservation(reservationRequest.toReservation());
-            return reservationDAO.findReservation(id);
+    public Reservation createReservation(ReservationCreateRequest reservationCreateRequest) {
+        if (reservationRepository.checkSchedule(reservationCreateRequest.getDate(), reservationCreateRequest.getTime()) != 0) {
+            throw new DuplicateReservationScheduleException("중복된 예약 발생");
         }
-        return null;
+        Theme theme = themeRepository.findThemeById(reservationCreateRequest.getThemeId()).orElseThrow(ThemeNotFoundException::new);
```

Comment:
하나의 라인에 하나의 .을 사용하면 가독성이 좋아집니다 :)
인틸리제이로 브레이킹포인트를 설정하고 디버깅할 때나 코드를 작성할 때 오른쪽에 타입을 알려주어서 읽기 편하게 되니 참고해주세요~

```suggestion
        Theme theme = themeRepository.findThemeById(reservationCreateRequest.getThemeId()).
                orElseThrow(ThemeNotFoundException::new);
```